prime-router/docs/universal-pipeline/README.md

```diff
@@ -1,24 +1,262 @@
+
+> The client (sender in our case) is responsible for polling the history endpoint to ensure either success or failure of their message. ReportStream will not automatically alert senders of changes to their Report's status.
+
+#### History Endpoint
+
+As a Report flows through the Universal Pipeline, metadata such as the following is captured and stored in the internal Postgres database:
+- The name of the sender
+- The name of the receiver(s)
+- Whether the report was successfully sent
+- Any errors or warnings associated with a report
+- Was a report split up at any point in the pipeline? If so, what are the new reports?
+
+When the history endpoint is called, it will return a JSON object representing the aforementioned metadata stored in the database. This is the same object that is returned by the reports or waters endpoints when submitting a report, only it may contain more information that is not known at the time a report is submitted, like the name of the receiver(s). To better understand the history endpoint, see the [example below](#history-endpoint-example).
+
+> Make sure to check out the [Swagger documentation](../design/swagger.md) to learn how to execute the various endpoints provided by ReportStream and view their documentation, including the history endpoint!
+
+##### History Endpoint Example
+
+Step 1: User submits a report via the reports API and gets the following object back in the response. There are two very important things to note here:
+1. The `overallStatus` is `Received`, meaning RS accepted the file and it is awaiting further processing
+2. The `id` is a generated UUID that uniquely identifies the *submitted* report. This ID is used in the history query
+
+```JSON
+{
+	"id": "5036f90b-edd5-4a93-9edd-a71d1fa8fba1",
+	"submissionId": 980,
+	"overallStatus": "Received",
+	"timestamp": "2023-07-25T16:42:26.566Z",
+	"plannedCompletionAt": null,
+	"actualCompletionAt": null,
+	"sender": "CDC-ELIMS-HL7.CDC-ELIMS-SENDER",
+	"reportItemCount": 1,
+	"errorCount": 0,
+	"warningCount": 0,
+	"httpStatus": 201,
+	"destinations": [],
+	"actionName": "receive",
+	"externalName": null,
+	"reportId": "5036f90b-edd5-4a93-9edd-a71d1fa8fba1",
+	"topic": "full-elr",
+	"errors": [],
+	"warnings": [],
+	"destinationCount": 0
+}
+```
+
+Step 2: Some time later, could be right away or could be days from the original submission, the user calls the history endpoint with the id returned from the submission query (`5036f90b-edd5-4a93-9edd-a71d1fa8fba1`) and gets the following object back in the response. The important things to note here are:
```

Comment:
I am torn between "This is too much detail for an overview" and "this would have been nice to know when onboarding and helps tie it all together." for this part.

Reply:
I think this is appropriate here so the reader can better understand the core user-facing functionality of the UP - report status tracking. This is something senders are really interested in.